plugins/defaultplugins/ifplugin/interface_config_test.go

```diff
@@ -0,0 +1,258 @@
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/ip"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/memif"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/tap"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/vpe"
+	if_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+	"github.com/ligato/vpp-agent/tests/vppcallmock"
+	. "github.com/onsi/gomega"
+)
+
+var ifNames = []string{"if1", "if2", "if3"}
+var mtu uint32 = 1500
+var netAddresses = []string{"10.0.0.1/24", "10.0.0.2/24", "192.168.50.1/24"}
+var ipv6Addresses = []string{"fd21:7408:186f::/48", "2001:db8:a0b:12f0::1/48"}
+var macs = []string{"46:06:18:DB:05:3A", "BC:FE:E9:5E:07:04"}
+
+/* Interface configurator init and close */
+
+// Test init function
+func TestInterfaceConfiguratorInit(t *testing.T) {
+	// Setup
+	RegisterTestingT(t)
+	ctx := &vppcallmock.TestCtx{
+		MockVpp: &mock.VppAdapter{},
+	}
+	connection, err := core.Connect(ctx.MockVpp)
+	Expect(err).To(BeNil())
+	plugin := &InterfaceConfigurator{
+		Log:      logrus.DefaultLogger(),
+		GoVppmux: connection,
+	}
+	swIfIndices := ifaceidx.NewSwIfIndex(nametoidx.NewNameToIdx(plugin.Log, "swIf-test", "swIf", nil))
+	dhcpIndices := ifaceidx.NewDHCPIndex(nametoidx.NewNameToIdx(plugin.Log, "dhcp-test", "dhcp", nil))
+	ifVppNotifChan := make(chan govppapi.Message, 100)
+	// Reply set
+	ctx.MockVpp.MockReply(&memif.MemifSocketFilenameDetails{
+		SocketID:       1,
+		SocketFilename: []byte("test-socket-filename"),
+	})
+	ctx.MockVpp.MockReply(&vpe.ControlPingReply{})
+	// Register
+	swIfIndices.RegisterName(ifNames[0], 1, nil)
+	swIfIndices.RegisterName(ifNames[1], 2, nil)
+	// Test init
+	err = plugin.Init(swIfIndices, dhcpIndices, mtu, ifVppNotifChan)
+	Expect(err).To(BeNil())
+	Expect(plugin.swIfIndexes).ToNot(BeNil())
+	Expect(plugin.dhcpIndices).ToNot(BeNil())
+	Expect(plugin.notifChan).ToNot(BeNil())
+	Expect(plugin.mtu).To(BeEquivalentTo(mtu))
+	Expect(plugin.afPacketConfigurator).ToNot(BeNil())
+	Expect(plugin.memifScCache["test-socket-filename"]).To(BeEquivalentTo(1))
+	// Test DHCP notifications
+	dhcpIpv4 := &dhcp2.DhcpComplEvent{
+		HostAddress:   net.ParseIP(ipAddresses[0]),
+		RouterAddress: net.ParseIP(ipAddresses[1]),
+		HostMac: func(mac string) []byte {
+			parsed, err := net.ParseMAC(mac)
+			Expect(err).To(BeNil())
+			return parsed
+		}("7C:4E:E7:8A:63:68"),
+		Hostname: []byte(ifNames[0]),
+		IsIpv6:   0,
+	}
+	dhcpIpv6 := &dhcp2.DhcpComplEvent{
+		HostAddress:   net.ParseIP(ipv6Addresses[0]),
+		RouterAddress: net.ParseIP(ipv6Addresses[1]),
+		HostMac: func(mac string) []byte {
+			parsed, err := net.ParseMAC(mac)
+			Expect(err).To(BeNil())
+			return parsed
+		}("7C:4E:E7:8A:63:68"),
+		Hostname: []byte(ifNames[1]),
+		IsIpv6:   1,
+	}
+	plugin.dhcpChan <- dhcpIpv4
+	plugin.dhcpChan <- dhcpIpv6
+	time.Sleep(1 * time.Second)
```

Comment:
Don't use sleeps in tests.. rather use [Eventually](https://onsi.github.io/gomega/#eventually).